CodeEdit/Features/SourceControl/SourceControlManager+GitClient.swift

```diff
@@ -188,7 +188,10 @@ extension SourceControlManager {
 
     /// Commit files selected by user
     func commit(message: String, details: String? = nil) async throws {
-        try await gitClient.commit(message: message, details: details)
+        guard let repository else { return }
```

Comment:
I find guarding on `repository` in these places confusing. I understand that we need to be able to unwrap this optional object. But I don't think the call sites of this function should ever be shown unless a valid `repository` exists. 

This feels like a great use of a lock and key pattern. Here is a good conference [talk](https://youtu.be/bAokeFxW2fo?si=eil-AmZyq8qGKxop&t=790) that goes over it

Reply:
You are right, I just put it there for now because I am not sure the optimal solution for optional repository property.